docs/api/covid_hosp.md

```diff
@@ -11,11 +11,19 @@ Hospital Capacity by State" datasets provided by the US Department of
 Health & Human Services via healthdata.gov. The latter provides more frequent updates,
 so it is combined with the former to create a single dataset which is as recent as possible.
 
+HHS performs up to four days of forward-fill for missing values. One day of forward-fill
```

Comment:
I'm not exactly sure how to interpret "affecting 28 states in each issue on average", and I'm having trouble reproducing.  [Maybe because I was trying on influenza admissions rather than covid, or checking `as_of` data rather than `issue` update data?]  I've tried `as_of` each issue in `full_seq(<issue data>$issue, 1L)`, and checked for each state whether it has a repeated value anywhere in its time series, and that seems to flag everywhere.

Adapting some stats [here](https://delphi-org.slack.com/archives/C028AUR6MV1/p1666210959243899) to run on all `as_of` data (for all dates in the time range for which we have had influenza issue data), which might be more informative:
<details> <summary>Code + results:</summary>

```r
library(epidatr)
library(epiprocess)
library(data.table)
library(dplyr)
library(ggplot2)

confirmed_admissions_influenza_1d_tbl = covidcast(
  data_source = "hhs",
  signals = "confirmed_admissions_influenza_1d",
  time_type = "day",
  geo_type = "state",
  time_values = epirange(12340101, 34560101),
  geo_values = "*",
  # for better reproducibility: stop with issue data for yesterday
  issues = epirange(12340101, 20221026)
) %>% fetch_tbl()

confirmed_admissions_influenza_1d_archive = confirmed_admissions_influenza_1d_tbl %>%
  rename(version = issue) %>%
  as_epi_archive()

ref_time_values = tidyr::full_seq(confirmed_admissions_influenza_1d_archive$DT$version, 1L)

influenza_analysis_by_version_lag =
  confirmed_admissions_influenza_1d_archive %>%
  epix_slide(function(edf, grp) {
    edf %>%
      arrange(geo_value, time_value) %>%
      mutate(no_change =
               lag(.data$time_value) == .data$time_value - 1L &
               lag(.data$value) == .data$value) %>%
      group_by(version_lag = as.integer(attr(., "metadata")$as_of - .data$time_value)) %>%
      summarize(
        n_no_change = sum(!is.na(.data$no_change) & .data$no_change),
        n_na_change = sum(is.na(.data$no_change)),
        n = n(),
        .groups = "drop"
      ) %>%
      {tibble(stats = list(.))}
  }, n = 60L, group_by=c(), ref_time_values = ref_time_values) %>%
  group_by(time_value) %>%
  slice_head(n = 1L) %>%
  ungroup() %>%
  unnest(slide_value_stats) %>%
  group_by(version_lag) %>%
  summarize(across(starts_with("n"), sum),
            .groups = "drop") %>%
  # XXX makes cols with suboptimal names:
  mutate(across(starts_with("n_"), list(prop=~.x/n)))

influenza_analysis_by_version_lag %>%
  print(n=10L)
#> # A tibble: 59 × 6
#>    version_lag n_no_change n_na_change     n n_no_change_prop n_na_change_prop
#>          <int>       <int>       <int> <int>            <dbl>            <dbl>
#>  1           1        2037          58  2105            0.968         0.0276  
#>  2           2       19645          63 20925            0.939         0.00301 
#>  3           3        9693          62 21539            0.450         0.00288 
#>  4           4        8845           9 21599            0.410         0.000417
#>  5           5        8331         224 21600            0.386         0.0104  
#>  6           6        8197         225 21600            0.379         0.0104  
#>  7           7        8394          12 21600            0.389         0.000556
#>  8           8        8392          12 21600            0.389         0.000556
#>  9           9        8410          12 21600            0.389         0.000556
#> 10          10        8416          12 21600            0.390         0.000556
#> # … with 49 more rows
#> # ℹ Use `print(n = ...)` to see more rows
```
</details>

I think this version-lag view is a bit more informative, assuming I've done it correctly, but might still be hard to describe precisely.  Perhaps we could can just skip exact stats and say that this contributes to a very common situation where admissions for a given `geo_value` and `time_value` are initially reported as the same as they were for that `geo_value` at `time_value-1`, but are later revised to a different value.

Reply:
I'm with you that it's probably better to avoid exact stats. My analysis differs from yours but since we can get significantly different results by still making reasonable decisions, I'm loath to try and justify one particular analysis as correct.

Mine:
* uses the covid_hosp endpoint, which anchors issue dates to healthdata.gov (covidcast issue dates often differ from healthdata.gov)
* is limited to first 100 issues in 2022
* counts all columns in the dataset, not just confirmed influenza admissions
* only counts where the last 2-4 reference dates in the issue have the same values, not any sequence of repeated values... but now I'm rethinking, and I suspect I misunderstood the specific effect we're worried about

<details>
<summary>Show code and results</summary>

```
import requests
import pandas as pd
from datetime import date, timedelta

URL = "https://delphi.cmu.edu/epidata/covid_hosp_state_timeseries/"
HEADERS = {
	"user-agent": "krivard-covid_hosp-clvfbot/1.0"
}
ONE_DAY = timedelta(days=1)
NCOL=114

def params(dates, issues):
	return {
		"states": "al,ak,az,ar,ca,co,ct,de,dc,fl,ga,hi,id,il,in,ia,ks,ky,la,me,md,ma,mi,mn,ms,mo,mt,ne,nv,nh,nj,nm,ny,nc,nd,oh,ok,or,pa,ri,sc,sd,tn,tx,ut,vt,va,wa,wv,wi,wy,as,fm,gu,mh,mp,pw,pr,um,vi",
		"dates": dates,
		"issues": issues
	}

def go(params):
	req = requests.get(URL, params, headers=HEADERS)
	req.raise_for_status()
	ret = pd.DataFrame(req.json()['epidata'])
	if "state" in ret.columns and "date" in ret.columns:
		return ret.set_index(["state","date"]).sort_index()
	else:
		print(ret)
		return None

def dtoi(d):
	return int(d.strftime("%Y%m%d"))

def last_four(ending):
	return ",".join(
		str(dtoi(ending - i*ONE_DAY))
		for i in range(4)
	)

def fetch(issue):
	return go(params(last_four(issue), dtoi(issue)))

def make(d=date(2022, 1, 31)):
	df = fetch(d)
	if df is None: return None
	di = dtoi(d)
	de = df.xs(di, level="date")
	cursor = de == de # was: de != 0
	dt = []
	tmp_de = False
	for i in range(1,4):
		dti = dtoi(d-i*ONE_DAY)
		dtif = df.xs(dti, level="date")
		if de.shape != dtif.shape:
			if de.shape[0] > dtif.shape[0]:
				# dtif has fewer states; temporarily knock the extras out of de and we'll put them back later
				tmp_de = de
				de = de.loc[~de.index.isin(de.index.difference(dtif.index))]
			else:
				dtif = dtif.loc[~dtif.index.isin(dtif.index.difference(de.index))]
		di = de == dtif
		cursor = cursor & di
		dt.append(cursor.sum(axis=1))
		if tmp_de:
			# restore de to its full glory
			de = tmp_de
			tmp_de = False
	return pd.concat(dt, axis=1, keys=range(1,4))

START = date(2022,1,1)
all = {}
for i in range(100):
	d = START + i*ONE_DAY
	print(".", end="")
	mi = make(d)
	all[dtoi(d)] = mi
	if mi is None:
		print(f"Bad issue {d}")

mega = pd.concat(all, names=["issue"]).reorder_levels("state","issue")

fs = {}
for days in [1, 2, 3]:
	for columns in [114, 113, 112, 111, 110]:
		fs[(days, columns)] = (mega[[days]] == columns).sum(axis=1).groupby(by="state").sum()
filled = pd.concat(fs, names=["days", "columns"])

# >>> filled
# days  columns  state
# 1     114      AK       0
#                AL       0
#                AR       0
#                AS       0
#                AZ       0
#                        ..
# 3     110      VT       0
#                WA       0
#                WI       0
#                WV       0
#                WY       0
# Length: 810, dtype: int64
# >>> mega
#                   1    2    3
# state issue                  
# AK    20220101  113   86   68
# AL    20220101  113   18   10
# AR    20220101  113   64   40
# AS    20220101  113  113  113
# AZ    20220101  113   36    5
# ...             ...  ...  ...
# VT    20220410  107   69   59
# WA    20220410   57   34   13
# WI    20220410  113   63   59
# WV    20220410  113   96   61
# WY    20220410  113   87   79
# 
# [5227 rows x 3 columns]
# >>> (mega[[1]] == 113).sum(axis=1).groupby(by="issue").sum().mean()
# 28.628865979381445
# >>> (mega[[2]] == 113).sum(axis=1).groupby(by="issue").sum().mean()
# 0.18556701030927836
# >>> (mega[[2,3]] == 113).sum(axis=1).groupby(by="issue").sum().mean()
# 0.21649484536082475
```

</details>

or,
* Number of states with last two values the same (1 repeat) across all (113) columns, in each issue, averaged over 100 issues: 29
* Number of states with last three or last four values the same (2 or 3 repeats) across all (113) columns, in each issue, averaged over 100 issues: 0.2

Reply:
Agreed.  I like that the current analysis hits on the major points regarding length & revision of the repeat sequences.